kernel/src/kernel.rs

```diff
@@ -116,17 +120,29 @@ fn try_allocate_grant<KR: KernelResources<C>, C: Chip>(
         })
 }
 
+struct KernelProcessInitCapability {}
+unsafe impl capabilities::ProcessInitCapability for KernelProcessInitCapability {}
+
 impl Kernel {
-    pub fn new(processes: &'static [Option<&'static dyn process::Process>]) -> Kernel {
+    pub fn new(
+        processes: &'static [Option<&'static dyn process::Process>],
+        verifier: Option<&'static dyn AppVerifier>,
```

Comment:
We should change the Kernel to accept a `Verifier` using a generic type parameter, instead of a runtime option. The `()` (unit) type would then represent no verifier being used. This typically allows for massive optimizations, because Rust can determine that all function invocations on the passed verifier are no-ops, whereas with dynamic dispatch it generally cannot.

Reply:
I thought about this, and didn't go that direction mostly due to ergonomics. I will try it and see how it looks.

Reply:
Adding a generic type parameter to Kernel is pretty rough -- it cascades into every type that has a reference to Kernel (e.g., ProcessId, Grant, etc.). 

I think Hudson's comment points at a better approach: have a method that lets you load processes without verification. You can still call `load_and_check_processes` with `None`, to allow simple configuration of the checking policy, but if you want to completely elide the checking state machine you call this alternative method. It looks like doing this saves about 1kB of code.

Reply:
Yeah, this was my suggestion, glad it sounds workable. Still saving 1kB of code sounds like there is a decent amount of size overhead remaining -- like 1.5kB?

Reply:
I think `verifier` should be part of the `KernelResources` trait. That is the canonical way to configure the kernel, and how to handle app verification seems like a kernel configuration. Having to pass in a series of `None`s when creating the kernel struct is hard to document and not the most ergonomic, which is why we created the kernel resources trait. Or, if that isn't possible for some reason, we should be able to clearly articulate why not and what makes the verifier different.